### PR TITLE
fix: add errors to ticket when errored job is cancelled

### PR DESCRIPTION
**Pull request by pgallik**

### CODE REVIEW

File: src/Grb.Building.Api/Handlers/CancelJobHandler.cs [hunk 1]
```diff
@@ -48,9 +46,17 @@ public async Task Handle(CancelJobRequest request, CancellationToken cancellatio
                 ThrowCancelException();
             }
 
-            await _ticketing.Complete(
-                job.TicketId!.Value,
-                new TicketResult(new { JobStatus = "Cancelled" }),
+            var jobRecordErrors = _buildingGrbContext
```
Comment (jvandaal) [inline src/Grb.Building.Api/Handlers/CancelJobHandler.cs:49]: Job has no records. Else we cannot cancel it.

Comment (jvandaal) [reply src/Grb.Building.Api/Handlers/CancelJobHandler.cs:49]: Error should be retrieved from the ticket.

File: src/Grb.Building.Api/Handlers/CancelJobHandler.cs [hunk 2]
```diff
@@ -38,19 +38,23 @@ public async Task Handle(CancelJobRequest request, CancellationToken cancellatio
...
                 ThrowCancelException();
             }
 
+            var ticket = await _ticketing.Get(job.TicketId!.Value, cancellationToken);
+
             await _ticketing.Complete(
                 job.TicketId!.Value,
-                new TicketResult(new { JobStatus = "Cancelled" }),
+                ticket!.Result is not null && ticket.Status == TicketStatus.Error
+                    ? new TicketResult(new {JobStatus = "Cancelled", Result = System.Text.Json.JsonSerializer.Deserialize<TicketError>(ticket.Result.ResultAsJson)})
```
Comment (jvandaal) [inline src/Grb.Building.Api/Handlers/CancelJobHandler.cs:56]: Change name Result property into Error?

**Review by jvandaal — APPROVED** [review]

**Review by ArneD — APPROVED** [review]

**Pull request merged**